src/proto/console/text/output.rs

```diff
@@ -118,6 +119,58 @@ impl Output {
     }
 }
 
+impl fmt::Write for Output {
+    fn write_str(&mut self, s: &str) -> fmt::Result {
+        // Allocate a small buffer on the stack.
+        const BUF_SIZE: usize = 128;
+        // Add 1 extra character for the null terminator.
+        let mut buf = [0u16; BUF_SIZE + 1];
+
+        let mut i = 0;
+
+        // This closure writes the local buffer to the output and resets the buffer.
+        let mut flush_buffer = |buf: &mut [u16], i: &mut usize| {
+            buf[*i] = 0;
+            *i = 0;
+
+            self.output_string(buf.as_ptr())
+                .map_err(|_| fmt::Error)
+        };
+
+        {
+            // This closure converts a character to UCS-2 and adds it to the buffer,
+            // flushing it as necessary.
+            let mut add_char = |ch| {
+                // UEFI only supports UCS-2 characters, not UTF-16,
+                // so there are no multibyte characters.
+                buf[i] = ch;
+                i += 1;
+
+                if i == BUF_SIZE {
+                    flush_buffer(&mut buf, &mut i).map_err(|_| ucs2::Error::BufferOverflow)
+                } else {
+                    Ok(())
+                }
+            };
+
+            let add_ch = |ch| {
+                add_char(ch)?;
+
+                if ch == '\n' as u16 {
+                    add_char('\r' as u16)
```

Comment:
IIUC, this code will emit "\n\r" instead of the more usual "\r\n". Mind if I adjust that?

Reply:
Sure. Don't know if it changes stuff in practice.